custom/src/main/java/co/elastic/otel/dynamicconfig/DynamicConfiguration.java

```diff
@@ -131,6 +134,44 @@ public void stopDisablingAllTraces() {
     reenableTracesFor(ALL_INSTRUMENTATION);
   }
 
+  // okay to synchronize as this should only be called after multi-second intervals and
+  // additionally only called from threads which are not doing anything application blocking
+  public synchronized void deactivateInstrumentations(String deactivateList) {
```

Comment:
[minor] could we delegate the implementation to a static package-private method and test it so that it would just compute the two `to activate` and `to disable` sets and update the `alreadyDeactivated` collection only in the public non-static method ?

Reply:
It's already being tested via the integration smoke test which uses the active DynamicConfigurationPropertyChecker - which in turn needs this to be working correctly

I think splitting it how you suggest would make it more complicated, but I'll play and see

Reply:
updated